codama_parser_generators/meteora.cjs

```diff
@@ -0,0 +1,48 @@
+const path = require("node:path");
+const { rootNode } = require("@codama/nodes");
+const { rootNodeFromAnchor } = require("@codama/nodes-from-anchor");
+const { readJson } = require("@codama/renderers-core");
+const { visit } = require("@codama/visitors-core");
+
+const { renderVisitor: renderRustVisitor } = require("@codama/renderers-rust");
+const {
+  renderVisitor: renderVixenVisitor,
+} = require("@codama/renderers-vixen-parser");
+
+const project = "meteora";
+// const project = "pump_fun";
+
+const sdkName = project + "_sdk";
+const rootDir = process.cwd();
+
+const idl = readJson(
+  path.join(rootDir, "crates/parser/src", project, "idl.json")
+);
+
+// const node = rootNode(idl.program);
+const node = rootNodeFromAnchor(idl);
+
+// #Renderers-rust
+visit(
+  node,
+  renderRustVisitor(
+    path.join(rootDir, "crates/codama_sdks", sdkName, "src", "generated"),
+    {
+      crateFolder: path.join(rootDir, "crates/codama_sdks", sdkName),
+      formatCode: true,
+    }
+  )
+);
+
+//  #Render Vixen Parser
+visit(
+  node,
+  renderVixenVisitor(
+    path.join(rootDir, "crates/parser/src", project, "generated"),
+    {
+      sdkName,
+      crateFolder: path.join(rootDir, "crates/parser/src", project),
+      formatCode: true,
+    }
+  )
+);
```

Comment:
When I was running generator for yellowstone-shield I didn't get any conflict from having the parser and client in the same crate. For these parsers I'd like to try that so we have

```
/crates
  /parsers
    /meteora
      src/
        generated/*
      Cargo.toml
idls/
  meteroa.json
  pumpfun.json
package.json
parsers.mjs
```

We have a single `codama.mjs` script that loops through all IDLs in the `idls` folder. You run through `npm run generate`

I think the sdkName in this case is either `crate` or `super`. I hope it works like that or its just the same name as the parser

Reply:
Switching to packaging idl and codama.mjs with each parser crate.